crates/nu-cli/src/completions/completion_options.rs

```diff
@@ -261,6 +307,11 @@ mod test {
     #[case(MatchAlgorithm::Fuzzy, "example text", "ext", true)]
     #[case(MatchAlgorithm::Fuzzy, "example text", "mplxt", true)]
     #[case(MatchAlgorithm::Fuzzy, "example text", "mpp", false)]
+    #[case(MatchAlgorithm::Substring, "example text", "", true)]
+    #[case(MatchAlgorithm::Substring, "example text", "examp", true)]
+    #[case(MatchAlgorithm::Substring, "example text", "ple", true)]
+    #[case(MatchAlgorithm::Substring, "example text", "text", true)]
+    #[case(MatchAlgorithm::Substring, "example text", "mplxt", false)]
```

Comment:
Nitpick: We definitely want to be thorough with testing, and it's good that you added these, but I'm not sure all of these test cases are necessary. Do let me know if you have a reason for including them all, though, I might be missing something that you ran into.